ingestion/src/metadata/ingestion/api/workflow.py

```diff
@@ -214,3 +214,6 @@ def print_status(self) -> int:
         else:
             click.secho("Workflow finished successfully", fg="green", bold=True)
             return 0
+
+    def set_spark(self, spark):
+        self.source.set_spark(spark)
```

Comment:
Maybe stupid question, why do we need this set_spark at workflow level? 🤔

Reply:
it feels like spilling source implementation logic, which ideally should not happen

Reply:
Hey @pmbrul, I had added the `set_spark` method because this enables user to create his own sparks session and use the same for ingestion 

Shall I remove this 🤔 ?

Reply:
ideally this should be set in source itself.

Reply:
Let's move it out from Workflow now. If needed, we can update the JSON Schema of the sourceConfig to have an optional `SparkSessionURL` or something like that. I'll open an issue to tackle this in the future, but if it now works with an ad-hoc session, then its fine. Thanks

Reply:
we can add the same in prepare itself.